### ISSUE TITLE
New nitro meter broken in nitro challenges

### ISSUE THREAD

**Issue by Arthur-D** [supertuxkart/stk-code#3195]
## Description
With the new speedometer/nitro meter I get graphical glitches when doing nitro challenges. Tried debugging with @Alayan-stk and he cannot reproduce it, and the debug prints to show screen coordinates are correct, so something strange is going on. See attached screenshot and the YouTube link.
![sandtrack-2018 04 13_23 31 44](https://user-images.githubusercontent.com/5515261/38781353-c0e03c32-40e3-11e8-8efe-28263d634d35.jpg)

Video: https://youtu.be/9hprDtivYaQ

## Steps to reproduce
Play a nitro challenge in story mode. You might not be able to reproduce it anyway, but if you do, please reply that you did.

If you did not download STK from the official website, please specify where you got it:
git

System: 
Kubuntu 17.10
Graphics card: 
NVIDIA GTX 960, proprietary driver version 387.34
CPU: 
Intel i5-3570K

**Comment by Alayan-stk-2** [comment]
We did extensive debugging to find what's wrong. As @Arthur-D specified, the coordinates to draw the gauge are right when the issue happens.

There was an issue of a fixed array whose size hadn't been updated to be set to "vertices_count" rather than the previous direct value. I've fixed it in this branch : [https://github.com/STK-helper/stk-code/tree/FixGaugeI](url). But while this solved a part of the issue, what is shown in the picture and video remains unexplained for the moment.
